tests/classes/test_hypergraph.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+import xgi
+
+from xgi.classes.hypergraph import Hypergraph
+from xgi.exception import XGIError
+
+
+def test_constructor(edgelist5, dict5, incidence5, dataframe5):
+    el = edgelist5
+    d = dict5
+    i = incidence5
+    df = dataframe5
+
+    H_list = xgi.Hypergraph(el)
+    H_dict = xgi.Hypergraph(d)
+    H_mat = xgi.Hypergraph(i)
+    H_df = xgi.Hypergraph(df)
+
+    assert H_list.shape == H_dict.shape == H_mat.shape == H_df.shape
+    assert (
+        list(H_list.nodes)
+        == list(H_dict.nodes)
+        == list(H_mat.nodes)
+        == list(H_df.nodes)
+    )
+    assert (
+        list(H_list.edges)
+        == list(H_dict.edges)
+        == list(H_mat.edges)
+        == list(H_df.edges)
+    )
+    assert (
+        list(H_list.edges[0])
+        == list(H_dict.edges[0])
+        == list(H_mat.edges[0])
+        == list(H_df.edges[0])
+    )
+
+
+def test_name():
+    H = xgi.Hypergraph()
+    assert H.name == ""
+    H.name = "test"
+    assert H.name == "test"
+    H = xgi.Hypergraph(name="test")
+    assert H.name == "test"
+
+
+def test_contains(edgelist1):
+    el1 = edgelist1
+    H = xgi.Hypergraph(el1)
+    unique_nodes = list({node for edge in el1 for node in edge})
```

Comment:
Nitpick:
```suggestion
    unique_nodes = {node for edge in el1 for node in edge}
```

Reply:
Good point. I have no idea why I did it that way.